beacon_chain/networking/eth2_network.nim

```diff
@@ -2616,7 +2686,8 @@ proc broadcastAttestation*(
   let
     forkPrefix = node.forkDigestAtEpoch(node.getWallEpoch)
     topic = getAttestationTopic(forkPrefix, subnet_id)
-  node.broadcast(topic, attestation)
+  echo "about to send attestion"
+  node.broadcastTor(topic, attestation)
```

Comment:
The coexistence of these tor based calls and normal calls in the code would be refactored to avoid leakage risk you mentioned. The concern is that for Blocks/attestatations/proofs/slashing/discovery  etc.. should we move all over Tor if one of them is being transmitted over Tor? This needs to be examined further. As for code refactoring improvements, we can make tor methods private or in a separate file.

Reply:
Probably, moving most of them is required if any of them are moved. Ideally, yes, it's either all Tor or not at all, but it might be reasonable to avoid Tor for the largest and least frequent messages, such as blocks.

But attestations, discovery, et cetera all happen so regularly that it'd be easy to construct a correlation attack along the lines of https://github.com/vacp2p/nimbus-eth2-experimental/pull/3#discussion_r1319863398, yes.